test/FastTests/Client/RavenQueryTests.cs

```diff
@@ -0,0 +1,166 @@
+        public void CheckIfDocumentExists(Options options)
+        {
+            using (var store = GetDocumentStore(options))
+            {
+                new Employees_ByFirstName().Execute(store);
+
+                var employee1 = new NewEmployee { FirstName = "Golan", LastName = "Nahum", Address = new AddressInfo { Street = "123 Main St", } };
+                var employee2 = new NewEmployee { FirstName = "Grisha", LastName = "Kotler", Address = new AddressInfo { Street = "456 Main St" } };
+                {
+                    using (var session = store.OpenSession())
+                    {
+                        session.Store(employee1);
+                        session.Store(employee2);
+
+                        session.Advanced.WaitForIndexesAfterSaveChanges();
+                        session.SaveChanges();
+
+
+                        //Test 1:
+                        var queryUnnamedClass = session.Query<Employees_ByFirstName.IndexEntry, Employees_ByFirstName>()
+                            .Select(a => new { FirstName = a.FirstName, LastName = a.LastName, EmployeeId = Raven.Client.Documents.Queries.RavenQuery.Id(a) });
+
+                        var queryUnnamedClassString = queryUnnamedClass.ToString();
+                        var returnedEmployees1 = queryUnnamedClass.ToList();
+                        var returnedEmployee11 = returnedEmployees1.First();
+                        var returnedEmployee12 = returnedEmployees1.Last();
+
+                        Assert.Equal(
+                            "from index 'Employees/ByFirstName' select FirstName, LastName, id() as EmployeeId",
+                            queryUnnamedClassString);
+                        Assert.Equal(2, returnedEmployees1.Count);
+
+                        Assert.Equal(employee1.Id, returnedEmployee11.EmployeeId);
+                        Assert.Equal(employee1.FirstName, returnedEmployee11.FirstName);
+                        Assert.Equal(employee1.LastName, returnedEmployee11.LastName);
+                        Assert.Equal(employee2.Id, returnedEmployee12.EmployeeId);
+
+
+                        //Test 2:
+                        var queryUnnamedClass2 = session.Query<Employees_ByFirstName.IndexEntry, Employees_ByFirstName>()
+                            .Select(a => new { FullName = a.FirstName + " " + a.LastName, EmployeeId = Raven.Client.Documents.Queries.RavenQuery.Id(a) });
+
+                        var queryUnnamedClass2String = queryUnnamedClass2.ToString();
+                        var returnedEmployees2 = queryUnnamedClass2.ToList();
+                        var returnedEmployee21 = returnedEmployees2.First();
+                        var returnedEmployee22 = returnedEmployees2.Last();
+
+                        Assert.Equal(
+                            "from index 'Employees/ByFirstName' as a select { FullName " +
+                            ": a.FirstName+\" \"+a.LastName, EmployeeId : id(a) }",
```

Comment:
can we put it in 1 line without concatenation?

Reply:
Done.